docs/enhancements/replace_broken_clusters.md

```diff
@@ -0,0 +1,69 @@
+validate a cluster can be successfully resumed and is running as expected before assigning it to a claim.
+
+## Motivation
+
+As we move towards being more proactive with having clusters ready as [hot spares](https://github.com/openshift/hive/pull/1434),
+we could come across instances where stale hibernated clusters could fail to resume and still be assigned to ClusterClaims.
+There's a need to detect broken clusters and replace them automatically.
+
+### Goals
+
+Always strive to assign a running cluster to a claim
+
+### Non Goals
+
+Fixing clusters (or attached claims) once a running cluster has already been assigned.
+
+## Proposal
+
+### Stand-by clusters
+
+Currently, we consider all installed clusters which are not set to be deleted as Assignable clusters. 
+Introduce another bucket of stand by clusters, which are installed and are not set for deletion, but are hibernating.
+A cluster is only assignable if it is running. A hibernating cluster has to be resumed successfully to be added to the 
+assignable clusters bucket.
+Some of the factors that might inhibit successful resume of a cluster that has been hibernating for a while include  
+unapproved certificates and expired credentials. Introduction of a separate bucket for hibernating clusters 
+allows for a validation before it is assigned to a claim.
+
+### Transition to Assignable clusters
+
+If a claim is created and there are no assignable clusters, we would resume a hibernating cluster and check if the resume
+was successful. The validating condition could be the expected powerstate matches the actual powerstate. If we encounter
+an error, we can mark that cluster to be deleted and try to transition the next available stand-by cluster.
+Figuring out how long we wait for the resume process to complete can be tricky - as there are instances when we can wait
+indefinitely. So we can consider a timeout for that wait. Fine-tuning the said timeout would be dependent on the 
+platform and the infrastructure, hence left for the admins to decide. Default is no time-out - so waiting indefinitely 
+unless an error is encountered.
+
+#### Timeout in Hibernation controller
+
+Introduce an optional spec.powerstateTransitionTimeout in cluster deployments. Once the powerstate is changed and 
+hibernating condition reflects Resuming or Stopping reason, controller should allow for a maximum wait time counting 
+from the last transition time of the condition equal to this timeout before marking the cluster as broken. This can be 
+done via updating the hibernation condition to reflect the reason as TimedOutWaitingToResume or TimedOutWaitingToStop.
+Once ClusterPool controller comes across a broken cluster, it will mark it for deletion and create another cluster.
+
+## Risks and Mitigations
+
+### Timeout too little
+
+For non-fake clusters, it takes about 5 minutes for a cluster to be successfully resumed. We can include a validation 
```

Comment:
Nah, IMO that's going too far. Let the customer shoot themselves in the foot if they want.

Not to mention that we really couldn't say what a reasonable governor value should be. And imposing one could make this pretty hard to test :)